src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -365,91 +365,69 @@ ParsedQuery Visitor::visit(Parser::UpdateContext* ctx) {
 
 // ____________________________________________________________________________________
 ParsedQuery Visitor::visit(Parser::Update1Context* ctx) {
-  if (ctx->modify()) {
-    return visit(ctx->modify());
-  } else if (ctx->clear()) {
-    return visit(ctx->clear());
-  }
-
-  parsedQuery_._clause = parsedQuery::UpdateClause();
-
-  if (ctx->insertData() || ctx->deleteData()) {
-    // handles insertData and deleteData cases
-    visitIf(&parsedQuery_.updateClause().toInsert_, ctx->insertData());
-    visitIf(&parsedQuery_.updateClause().toDelete_, ctx->deleteData());
-  } else if (ctx->deleteWhere()) {
-    auto [toDelete, pattern] = visit(ctx->deleteWhere());
-    parsedQuery_.updateClause().toDelete_ = std::move(toDelete);
-    parsedQuery_._rootGraphPattern = std::move(pattern);
+  if (ctx->deleteWhere() || ctx->modify()) {
+    return visitAlternative<ParsedQuery>(ctx->deleteWhere(), ctx->modify());
   } else {
-    visitAlternative<void>(ctx->load(), ctx->drop(), ctx->add(), ctx->move(),
-                           ctx->copy(), ctx->create());
-    AD_FAIL();
+    parsedQuery_._clause = visitAlternative<parsedQuery::UpdateClause>(
+        ctx->load(), ctx->clear(), ctx->drop(), ctx->create(), ctx->add(),
+        ctx->move(), ctx->copy(), ctx->insertData(), ctx->deleteData());
   }
 
   return parsedQuery_;
 }
 
 // ____________________________________________________________________________________
-void Visitor::visit(const Parser::LoadContext* ctx) const {
-  reportNotSupported(ctx, "SPARQL 1.1 Update Load is");
+Load Visitor::visit(Parser::LoadContext* ctx) {
+  return Load{
+      static_cast<bool>(ctx->SILENT()), visit(ctx->iri()),
+      ctx->graphRef() ? visit(ctx->graphRef()) : std::optional<GraphRef>{}};
```

Comment:
1. you can use `std::nullopt` as the alternative in the ternary, it works.
2. If this happens more often, e could use a `visitOptional(ctx->graphRef()` function.

Reply:
`std::nullopt` does not work (I tried). The compiler complains that the different values have different types. The `visit` can alternatively be wrapped in a `std::make_optional`.
```
error: operands to ‘?:’ have different types ‘GraphRef’ {aka ‘ad_utility::triple_component::Iri’} and ‘const std::nullopt_t’
```